pkg/controllers/cassandra/testing/gen.go

```diff
@@ -0,0 +1,46 @@
+package testing
+
+import (
+	"fmt"
+	"math/rand"
+
+	"github.com/jetstack/navigator/pkg/apis/navigator/v1alpha1"
+)
+
+func FuzzCassandraNodePool(np *v1alpha1.CassandraClusterNodePool, rand *rand.Rand, size int) {
+	np.Replicas = rand.Int31n(5)
+}
+
+func FuzzCassandraClusterNodePools(cluster *v1alpha1.CassandraCluster, rand *rand.Rand, size int) {
+	if cluster.Spec.NodePools == nil {
+		cluster.Spec.NodePools = []v1alpha1.CassandraClusterNodePool{}
+	}
+	if cluster.Status.NodePools == nil {
+		cluster.Status.NodePools = map[string]v1alpha1.CassandraClusterNodePoolStatus{}
+	}
+	for i := 0; i < rand.Intn(5); i++ {
+		np := v1alpha1.CassandraClusterNodePool{
+			Name: fmt.Sprintf("np%d", i),
+		}
+		FuzzCassandraNodePool(&np, rand, size)
+		nps := v1alpha1.CassandraClusterNodePoolStatus{
+			ReadyReplicas: np.Replicas,
+		}
+		// 20% chance of ScaleOut
+		if rand.Intn(4) == 0 {
+			np.Replicas++
```

Comment:
Will increase it by 1 ever result in it being particularly large? Should we not just set this to some random value output by the rng?

Reply:
(maybe you are following some best practice I am not aware of though!)

Reply:
the initial value is between 0-5 (see code above) so it should never be more than 6.
This is a way to simulate an occasional difference between the Spec.NodePool.Replicas and Status.NodePools.ReadyReplicas